pkg/types/clusterconfig/errors.go

```diff
@@ -99,10 +100,17 @@ func ErrorIncompatibleSpotInstanceTypeGPU(target aws.InstanceMetadata, suggested
 	})
 }
 
+func ErrorIncompatibleInfSpotInstanceType(target aws.InstanceMetadata, suggested aws.InstanceMetadata) error {
```

Comment:
since this error message doesn't mention inferentia, we could either remove `Inf` from the error function / type, or mention inferentia in the error message

Reply:
Decided to follow the pattern of other resource-incompatible errors and rename it to `ErrorIncompatibleSpotInstanceTypeInf` (just like `ErrorIncompatibleSpotInstanceTypeCPU/Mem/GPU`). The error message has been renamed in that fashion.